src/pages/CommitDetailPage/Header/hooks/useCommitHeaderData.tsx

```diff
@@ -132,7 +138,7 @@ export const useCommitHeaderData = ({
         }
 
         return {
-          commit: data?.repository?.commit,
+          commit: data?.owner?.repository?.commit ?? null,
```

Comment:
so just so i understand this better, we can't return undefined here?

Reply:
It's because of the way that GraphQL works. If you request a field it will either return the value or `null` it will never return `undefined` so we don't wan't the TS value to reflect something it could never be. Which if we set to undefined here it won't pass the type validation that zod runs and will error out.